src/source_code_generator.rs

```diff
@@ -60,8 +58,8 @@ impl<'a> SourceCodeGenerator<'a> {
         }
     }
 
-    pub fn generate(self) -> Result<String, FromUtf8Error> {
-        String::from_utf8(self.buffer)
+    pub fn generate(self) -> String {
+        unsafe { String::from_utf8_unchecked(self.buffer) }
```

Comment:
This would be the only `unsafe` block in Ruff, which is unfortunate. Consider `.unwrap()` instead?

Reply:
Do you think it's a bad idea to fail hard here vs. propagating the error?

Reply:
Panicking with `unwrap` or `expect` seems like a fine reaction to a bug that broke our internal invariants; the backtrace would make it clear what happened in that event.